File: server/src/seed/inserters/learner_details.rs
use chrono::Utc;
use sea_orm::{DatabaseConnection, EntityTrait, Set};

use crate::seed::specs::LearnerDetailsSpec;
use crate::utils::AppError;
use ::entity::learner_details;

const CHUNK_SIZE: usize = 100;

pub async fn insert_learner_details(
    db: &DatabaseConnection,
    specs: &[LearnerDetailsSpec],
) -> Result<(), AppError> {
    if specs.is_empty() {
        return Ok(());
    }

    let now = Utc::now().naive_utc();

    let models: Vec<learner_details::ActiveModel> = specs
        .iter()
        .map(|spec| learner_details::ActiveModel {
            id: Set(spec.id),
            user_id: Set(spec.user_id),
            lrn: Set(spec.lrn.clone()),
            sex: Set(spec.sex.clone()),
            track_strand: Set(spec.track_strand.clone()),
            curriculum: Set(spec.curriculum.clone()),
            birthdate: Set(spec.birthdate),
            birthplace: Set(spec.birthplace.clone()),
            home_address: Set(spec.home_address.clone()),
            father_name: Set(spec.father_name.clone()),
            father_contact: Set(spec.father_contact.clone()),
            mother_name: Set(spec.mother_name.clone()),
            mother_contact: Set(spec.mother_contact.clone()),
            guardian_name: Set(spec.guardian_name.clone()),
            guardian_contact: Set(spec.guardian_contact.clone()),
            date_admitted: Set(spec.date_admitted),
            created_at: Set(now),
            updated_at: Set(now),
            deleted_at: Set(None),
        })
        .collect();

    for chunk in models.chunks(CHUNK_SIZE) {
        learner_details::Entity::insert_many(chunk.iter().cloned())
            .exec(db)
            .await
            .map_err(|e| AppError::InternalServerError(e.to_string()))?;
    }

    Ok(())
}

File: client/lib/domain/student_records/entities/learner_details.dart
import 'package:equatable/equatable.dart';

class LearnerDetails extends Equatable {
  final String id;
  final String userId;
  final String? lrn;
  final String? sex;
  final int? age;
  final String? trackStrand;
  final String? curriculum;
  final String? birthdate;
  final String? birthplace;
  final String? homeAddress;
  final String? fatherName;
  final String? fatherContact;
  final String? motherName;
  final String? motherContact;
  final String? guardianName;
  final String? guardianContact;
  final String? dateAdmitted;

  const LearnerDetails({
    required this.id,
    required this.userId,
    this.lrn,
    this.sex,
    this.age,
    this.trackStrand,
    this.curriculum,
    this.birthdate,
    this.birthplace,
    this.homeAddress,
    this.fatherName,
    this.fatherContact,
    this.motherName,
    this.motherContact,
    this.guardianName,
    this.guardianContact,
    this.dateAdmitted,
  });

  @override
  List<Object?> get props => [id];
}
